code/game/objects/structures/aliens.dm

```diff
@@ -82,23 +97,144 @@
-	base_icon_state = "resin_membrane"
-	opacity = FALSE
-	max_integrity = 160
-	resintype = "membrane"
+/*  Resin-Door
+	This borrs a bit of code from Mineral-Door, but is mostly its own thing due to not needing many of the latter's things instead of its own things */
+/obj/structure/alien/resin/door
+	name = "resin door"
+	density = TRUE
+	anchored = TRUE
+	opacity = TRUE
+
+	icon = 'icons/obj/smooth_structures/alien/resin_door.dmi'
+	icon_state = "resin"
+	base_icon_state = "resin"
+	max_integrity = 150
+	armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 0, BOMB = 0, BIO = 0, RAD = 0, FIRE = 50, ACID = 50)
+	damage_deflection = 0
+	flags_2 = RAD_PROTECT_CONTENTS_2 | RAD_NO_CONTAMINATE_2
+	rad_insulation = RAD_MEDIUM_INSULATION
+	var/initial_state
+	var/state_open = FALSE
+	var/is_operating = FALSE
+	var/close_delay = 15
+	smoothing_flags = null
+	var/openSound = 'sound/machines/alien_airlock.ogg'
+	var/closeSound = 'sound/machines/alien_airlock.ogg'
+
 	smoothing_groups = list(SMOOTH_GROUP_ALIEN_RESIN, SMOOTH_GROUP_ALIEN_WALLS)
 	canSmoothWith = list(SMOOTH_GROUP_ALIEN_WALLS)
 
-/obj/structure/alien/resin/CanPass(atom/movable/mover, turf/target)
-	if(istype(mover) && mover.checkpass(PASSGLASS))
-		return !opacity
+/obj/structure/alien/resin/door/Initialize()
+	. = ..()
+	initial_state = icon_state
+	QUEUE_SMOOTH_NEIGHBORS(src)
+	air_update_turf(1)
+	for(var/obj/structure/alien/weeds/node/W in get_turf(src))
+		qdel(W)
+	for(var/obj/structure/alien/weeds/E in get_turf(src))
+		return
+	new /obj/structure/alien/weeds(loc)
+
+/obj/structure/alien/resin/door/Destroy()
+	density = FALSE
+	QUEUE_SMOOTH_NEIGHBORS(src)
+	playsound(loc, pick('sound/effects/alien_resin_break2.ogg','sound/effects/alien_resin_break1.ogg'), 100, FALSE)
+	air_update_turf(1)
+	return ..()
+
+/obj/structure/alien/resin/door/Move()
+	var/turf/T = loc
+	. = ..()
+	move_update_air(T)
+
+/obj/structure/alien/resin/door/Crossed(mob/living/L, oldloc)
+	..()
+	if(!state_open)
+		return try_to_operate(L)
+
+/obj/structure/alien/resin/door/attack_ai(mob/user)
+	if(isAI(user))
+		return
+	if(isrobot(user))
+		return
+
+/obj/structure/alien/resin/door/attack_hand(mob/user)
+	return try_to_operate(user)
+
+/obj/structure/alien/resin/door/attack_ghost(mob/user)
+	if(user.can_advanced_admin_interact())
+		operate()
+
+/obj/structure/alien/resin/door/CanAtmosPass(turf/T)
 	return !density
 
+/obj/structure/alien/resin/door/proc/try_to_operate(atom/user)
+	if(is_operating)
+		return
+	if(iscarbon(user))
+		var/mob/living/carbon/C = user
+		if(C.get_int_organ(/obj/item/organ/internal/xenos/hivenode))
+			if(world.time - C.last_bumped <= 60)
+				return
+			if(!C.handcuffed)
+				operate()
+
+/obj/structure/alien/resin/door/proc/mobless_try_to_operate() // This 2nd try_to_operate() is needed so that CALLBACK can close the door without having to either call operate() and get bugged when clicked much or call try_to_operate(atom/user) and not be able to use it due to not having a mob using it
```

Comment:
```suggestion
/*
  * This 2nd try_to_operate() is needed so that CALLBACK can close the door without having to either call operate() and get bugged when clicked much or 
  * call try_to_operate(atom/user) and not be able to use it due to not having a mob using it
*/
/obj/structure/alien/resin/door/proc/mobless_try_to_operate() 
```
cleaner comment